packages/core/src/core/coreToolScheduler.ts

```diff
@@ -887,7 +888,8 @@ export class CoreToolScheduler {
       this.toolCalls = [];
 
       for (const call of completedCalls) {
-        logToolCall(this.config, new ToolCallEvent(call));
+        const event = new ToolCallEvent(call);
+        logToolCall(this.config, addProgrammingLanguageToEvent(event));
```

Comment:
programming language should be an attribute of the event itself. Can we refactor this to do that? Lets not change the top level calling function but instead have it be part of the event object.